File: KthLargestTerm.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        
        priority_queue<int,vector<int>,greater<int>> pq; //definition of min heap in c++
        
        for(int i=0;i<nums.size();i++)
        {
            if(pq.size()<k)
            {
                pq.push(nums[i]);
            }
            else if(nums[i]>pq.top())
            {
                pq.pop();
                pq.push(nums[i]);
            }
        }
        
        return pq.top();
    }
};

File: KthLargestTerm.java
package S30.Heaps_1;

import java.util.PriorityQueue;

/*
Time Complexity : O(NlogK)
Space Complexity : O(K)
Did this code successfully run on Leetcode : Yes
Any problem you faced while coding this : None
*/
public class KthLargestTerm {

    public int findKthLargest(int[] nums, int k) {

        if(nums == null || nums.length == 0) return -1;

        //constructing a min heap
        PriorityQueue<Integer> heap = new PriorityQueue<Integer>(
                (x,y) -> x - y
        );

        for(int i = 0; i < k; i++){
            heap.offer(nums[i]);
        }

        for(int i = k; i < nums.length; i++){
            if(heap.peek() < nums[i]){
                heap.offer(nums[i]);
            }
            if(heap.size() > k){
                heap.poll();
            }
        }

        return heap.poll();
    }
}
